src/triage/risklist/__init__.py

```diff
@@ -0,0 +1,171 @@
+from triage.component.results_schema import upgrade_db
+from triage.component.architect.entity_date_table_generators import EntityDateTableGenerator, DEFAULT_ACTIVE_STATE
+from triage.component.architect.features import FeatureGenerator
+from triage.component.architect.builders import MatrixBuilder
+from triage.component.catwalk.predictors import Predictor
+from triage.component.catwalk.utils import filename_friendly_hash
+from triage.util.conf import dt_from_str
+
+from collections import OrderedDict
+import json
+import re
+
+
+def get_required_info_from_config(db_engine, model_id):
+    """Get all information needed to make the risk list from model_id
+    Args:
+            db_engine (sqlalchemy.db.engine)
+            model_id (int) The id of a given model in the database
+
+    Returns: (dict) a dictionary of all information needed for making the risk list
+
+    """
+    get_experiment_query = """
+        select experiments.config, matrices.matrix_metadata, matrix_uuid
+        from model_metadata.experiments
+        join model_metadata.experiment_matrices using (experiment_hash)
+        join model_metadata.matrices using (matrix_uuid)
+        join model_metadata.models on (models.train_matrix_uuid = matrices.matrix_uuid)
+        where model_id = %s
+    """
+    results = list(db_engine.execute(get_experiment_query, model_id))
+    experiment_config = results[0]['config']
+    label_config = experiment_config['label_config']
+    original_matrix_uuid = results[0]['matrix_uuid']
+    matrix_metadata = json.loads(results[0]['matrix_metadata'])
+    feature_names = matrix_metadata['feature_names']
+    feature_config = experiment_config['feature_aggregations']
+    cohort_config = experiment_config['cohort_config']
+    timechop_config = experiment_config['temporal_config']
+    feature_start_time = timechop_config['feature_start_time']
+
+    model_info = {}
+    model_info['cohort_config'] = cohort_config
+    model_info['feature_config'] = feature_config
+    model_info['feature_names'] = feature_names
+    model_info['feature_start_time'] = feature_start_time
+    model_info['original_matrix_uuid'] = original_matrix_uuid
+    model_info['label_config'] = label_config
+
+    return model_info
+
+
+def generate_risk_list(db_engine, matrix_storage_engine, model_storage_engine, model_id, as_of_date):
+    """Generate the risk list based model_id and as_of_date
+
+    Args:
+            db_engine (sqlalchemy.db.engine)
+            model_id (int) The id of a given model in the database
+            matrix_storage_engine (catwalk.storage.matrix_storage_engine)
+            model_storage_engine (catwalk.storage.model_storage_engine)
+            as_of_date (string) a date string like "YYYY-MM-DD"
+    """
+    upgrade_db(db_engine=db_engine)
```

Comment:
Does the `upgrade_db` belong here unconditionally?

Reply:
It's alembic's 'upgrade HEAD' so it'll be idempotent. I think having it be unconditional is how we do it in the Experiment, so it would make sense to mirror that here.